ParseCNVQC/parse_cnv_qc.py

```diff
@@ -62,6 +62,51 @@ def make_mail(today, daysago, attachment, run_status):
     send_email(settings.email_from, settings.email_to, subject, text, attachment)
 
 
+def get_number_samples_per_run_from_samplesheet(folder, rawfolder, projects, warnings):
+    """
+    Determine number of samples in a run based on the SampleSheet.
+    Only includes samples that are in predefined projects as stated in settings file.
+    Includes a warning if Samplesheet.csv is not detected in the raw data folder
+
+    Args:
+        folder (string): full path to raw data folder
+        rawfolder (dict):
+            key: full path to raw data folder, values: list of all processed folder from run, total number of samples (int)
+        projects (list): projectIDs to include in total sample calculation. Excludes i.e. non-dx projects in calculations
+        warnings (list): List with warning messages
+
+    Returns:
+        rawfolder (dict): rawfolder including sample count
+        warnings (list): message list including potential new warnings
+    """
+    number_samples_run = 0
+    lanes = []
+    lane_index = ""
+    if os.path.exists("{}/SampleSheet.csv".format(folder)):
+        with open("{}/SampleSheet.csv".format(folder), 'r') as samplesheet:
+            sample_section = False
+            for line in samplesheet:
+                if sample_section:
+                    for project in projects:
+                        if project in line.upper():
+                            number_samples_run += 1
+                            if line.split(",")[lane_index] not in lanes:
+                                lanes.append(line.split(",")[lane_index])
+                if "Sample_ID" not in line:
+                    continue
+                else:
+                    sample_section = True
+                    header = [column for column in line.split(",")]
+                    lane_index = header.index('Lane')
+    else:
+        warnings.append("no samplesheet for run {}, assuming unknown number of samples in run".format(folder))
+
+    # prevent division by zero.
+    if len(lanes) > 0:
+        rawfolder[folder][1] += number_samples_run/len(lanes)
+    return rawfolder, warnings
```

Comment:
mocht er tijd voor zijn, hou dit stukje dan eens kritisch tegen het licht.

op dit moment zit er namelijk logica in de functie om het aantal samples in een samplesheet te bepalen, maar het doet ook dingen daar buitenom, b.v. een dict bijhouden met het aantal samples per project. eigenlijk mag dat best zijn eigen functie zijn.

en een mooi voorbeeldje van code die heerlijk zou werken met OO! 
dan kan je gewoon een run object maken dat 1 of meerdere projects heeft en de determine_samples uitvoeren op een run object, die het op alle (gefilterde) project lijst uitvoert